src/traces/scattergl/index.js

```diff
@@ -13,10 +13,12 @@ var ScatterGl = {};
 ScatterGl.attributes = require('./attributes');
 ScatterGl.supplyDefaults = require('./defaults');
 ScatterGl.colorbar = require('../scatter/colorbar');
+ScatterGl.hoverPoints = require('../scatter/hover');
 
 // reuse the Scatter3D 'dummy' calc step so that legends know what to do
-ScatterGl.calc = require('../scatter3d/calc');
+ScatterGl.calc = require('../scatter/calc');
```

Comment:
Ok. After some more thorough speed tests, making `scattergl` pass through `scatter/calc.js` adds roughly 1 full second to `Plotly.plot` at 1e6 points (clocking in the 2500-3000ms range as opposed to ~1500ms on `master`) on my machine. Moreover, reusing the calcdata-based cartesian hover routines make picking noticeably slower at 1e6 points. 

In brief, we can't go through with this approach. If `scattergl` is slow with 1e6 points, it loses its _raison d'être_.

It worth mentioning that at 1e5, the modifications in this branch perform very (and to my surprise) well. Still, millions of points is what `scattergl` is all about.

So, what should we do? I'm thinking we should start by reusing [`cartesian/select.js`](https://github.com/plotly/plotly.js/blob/master/src/plots/cartesian/select.js) and make its output listen to events on the `gl2d` mouse container (that this PR removed :smirk: ) and draw the SVG lasso/rectangle path above the gl canvas. From there, it shouldn't be too hard to modify `scattergl/select.js` to use the selection indices to dig in `fullData.x` and `fullData.y` (as the calcdata won't match the coordinates w/o calling the slow `scatter/calc.js`). 

@dfcreative let me know if this makes sense or if you have any questions.